app/src/main/java/com/tomtom/sdk/examples/startnavigation/NavigationActivity.kt

```diff
@@ -0,0 +1,125 @@
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        initRouting()
+        planRoute(amsterdam, rotterdam)
+    }
+
+    private fun initRouting() {
+        routePlanner = OnlineRoutePlanner.create(context = this, apiKey = apiKey)
+        routeReplanner = OnlineRouteReplannerFactory.create(routePlanner)
+    }
+
+    private fun planRoute(origin: GeoPoint, destination: GeoPoint) {
+        val itinerary = Itinerary(origin = origin, destination = destination)
+        routePlanningOptions = RoutePlanningOptions(
+            itinerary = itinerary,
+            guidanceOptions = GuidanceOptions(
+                instructionType = InstructionType.Text,
+                phoneticsType = InstructionPhoneticsType.Ipa,
+                announcementPoints = AnnouncementPoints.All,
+                extendedSections = ExtendedSections.All,
+                progressPoints = ProgressPoints.All
+            ),
+            vehicle = Vehicle.Car()
+        )
+        routePlanner.planRoute(routePlanningOptions!!, routePlanningCallback)
+    }
+
+    private val routePlanningCallback = object : RoutePlanningCallback {
+        override fun onSuccess(result: RoutePlanningResponse) {
+            route = result.routes.first()
+            initLocationProvider()
+            initNavigation()
+            startNavigation()
+        }
+
+        override fun onFailure(failure: RoutingFailure) {}
+
+        override fun onRoutePlanned(route: Route) = Unit
+    }
+
+    private fun initLocationProvider() {
+        val routeGeoLocations = route!!.geometry.map { GeoLocation(it) }
+        val simulationStrategy = InterpolationStrategy(routeGeoLocations)
+        locationProvider = SimulationLocationProvider.create(strategy = simulationStrategy)
+        locationProvider.enable()
+    }
+
+    private fun initNavigation() {
+        val navigationConfiguration = NavigationConfiguration(
+            context = this,
+            apiKey = apiKey,
+            locationProvider = locationProvider,
+            routeReplanner = routeReplanner
+        )
+        tomTomNavigation = TomTomNavigationFactory.create(navigationConfiguration)
+    }
+
+    private fun startNavigation() {
+        tomTomNavigation.addProgressUpdatedListener(progressUpdatedListener)
+
+        val routePlan = RoutePlan(route!!, routePlanningOptions!!)
+        val navigationOptions = NavigationOptions(routePlan)
+        tomTomNavigation.start(navigationOptions)        
+    }
+
+    private val progressUpdatedListener = ProgressUpdatedListener { routeProgress ->
+        val distanceAlongRoute = routeProgress.distanceAlongRoute
+        val remainingTimeOnRoute = routeProgress.remainingTime
+        Log.v("NavigationActivity", "distanceAlongRoute: " + distanceAlongRoute)
+        Log.v("NavigationActivity", "remainingTimeOnRoute: " + remainingTimeOnRoute)
+    }
+
+    companion object {
+        val amsterdam = GeoPoint(
```

Comment:
I think this should be one-liner at the place where you use it

Reply:
Yes, in the code snippets I will definitely put it right before where it is used.